ttnmqtt/ttnmqtt.py

```diff
@@ -0,0 +1,203 @@
+        self.port = port
+        self.confirmed = confirmed
+        self.schedule = schedule
+
+    def obj2json(self):
+        json_msg = json.dumps(self.__dict__)
+        return str(json_msg)
+
+
+class MyEvents(Events):
+    __events__ = (
+        "uplink_msg",
+        "downlink_msg",
+        "connect",
+        "close")
+
+
+class MQTTClient:
+
+    def __init__(self, app_id, app_access_key, **kwargs):
+        self.__client = mqtt.Client()
+        self.__appID = app_id
+        self.__accessKey = app_access_key
+        self.__events = MyEvents()
+        self.__mqttAddress = ""
+        self.__discoveryAddress = ""
+        for k, v in kwargs.items():
+            if k == "mqtt_address":
+                self.__mqttAddress = v
+            if k == "discovery_address":
+                self.__discoveryAddress = v
+
+    def connect(self):
+        self.__client.on_connect = self._on_connect()
+        self.__client.on_publish = self._on_downlink()
+        self.__client.on_message = self._on_message()
+        self.__client.on_disconnect = self._on_close()
+
+        self.__client.username_pw_set(self.__appID, self.__accessKey)
+        if self.__mqttAddress == "":
+            creds = grpc.ssl_channel_credentials()
+            if self.__discoveryAddress == "":
+                channel = grpc.secure_channel(
+                    'discovery.thethings.network:1900',
+                    creds)
+            else:
+                channel = grpc.secure_channel(self.__discoveryAddress, creds)
+            stub = discovery_pb2_grpc.DiscoveryStub(channel)
+            req = discovery_pb2.GetByAppIDRequest()
+            req.app_id = self.__appID
+            res = stub.GetByAppID(req)
+            self.__mqttAddress = res.mqtt_address
+            mqtt_addr = split_address(self.__mqttAddress)
+            if mqtt_addr['port']:
+                self.__client.connect(
+                    mqtt_addr['address'],
+                    mqtt_addr['port'],
+                    60)
+            else:
+                self.__client.connect(mqtt_addr['address'], 1883, 60)
+        else:
+            mqtt_addr = split_address(self.__mqttAddress)
+            if mqtt_addr['port']:
+                self.__client.connect(
+                    mqtt_addr['address'],
+                    mqtt_addr['port'],
+                    60)
+            else:
+                self.__client.connect(mqtt_addr['address'], 1883, 60)
+
+        self.start()
+
+    def _on_connect(self):
+        def on_connect(client, userdata, flags, rc):
+            if(rc == 0):
+                client.subscribe('{}/devices/+/up'.format(self.__appID))
+                res = True
+            else:
+                res = False
+            if self.__events.connect:
+                self.__events.connect(res, client=self)
+        return on_connect
+
+    def _on_close(self):
+        def on_disconnect(client, userdata, rc):
+            if rc != 0:
```

Comment:
This is more of a general comment, since you seem to be making this error multiple times: you can avoid if statements if the results are booleans.

You write:
```py
if rc != 0:
  res = False
else:
  res = True
```
this is the same as writing:
```py
res = rc != 0
```

The body of `on_disconnect` can be written more succinctly as:
```py 
def on_disconnect(client, userdata, rc):
  if self.__events.close:
    self.__events.close(rc != 0, client=self)
```